scipy/stats/stats.py

```diff
@@ -2331,6 +2333,107 @@ def zmap(scores, compare, axis=0, ddof=0):
+    standard deviation of ``log(a)``. Mathematically the population
+    geometric standard deviation is evaluated as::
+
+        gstd = exp(std(log(a))
+
+    The geometric standard deviation describes the spread of a set of numbers
+    where `gmean` is the preferred mean. It is a multiplicative factor, and
+    therefore a dimensionless quantity.
+
+    Parameters
+    ----------
+    a : array_like
+        An array like object containing the sample data.
+    axis : int, tuple or None, optional
+        Axis along which to operate. Default is 0. If None, compute over
+        the whole array `a`.
+    ddof : int, optional
+        Degrees of freedom correction in the calculation of the
+        geometric standard deviation. Default is 1.
+    nan_policy : {'propagate', 'raise', 'omit'}, optional
+        Defines how to handle input containing invalid values. In this
+        context an invalid value is defined as any non-finite or non positive
+        value.
+
+            'propagate' returns nan,
+            'raise' throws a  ValueError
+            'omit' performs the calculations ignoring nan values.
+
+        Default is 'raise'.
+
+    Returns
+    -------
+    ndarray or float
+        An array of the geometric standard deviation. If `axis` is None or `a`
+        is a 1d array a float is returned.
+
+    Notes
+    -----
+    As the calculation requires use of logarithms the geometric standard
+    deviation only supports strictly positive values. Any non-positive or
+    infinite values will raise a `ValueError` unless ``ignore_invalid=True``.
+
+    The geometric standard deviation is sometimes confused with the exponent of
+    the standard deviation, ``exp(std(a))``. Instead the geometric standard
+    deviation is ``exp(std(log(a)))``.
+
+    The default value for `ddof` is different to the default value (0) used
+    by other ddof containing functions, such as ``np.std`` and ``np.nanstd``.
+
+    Examples
+    --------
+    Find the geometric standard deviation of a log-normally distributed sample.
+    Note the standard deviation of the distribution is one, on a log scale this
+    evaluates to ``exp(1)``.
+
+    >>> from scipy.stats import gstd
+    >>> np.random.seed(123)
+    >>> sample = np.random.lognormal(mean=0, sigma=1, size=1000)
+    >>> gstd(sample)
+    2.7217860664589946
+
+    Compute the geometric standard deviation of a multidimensional array and
+    of a given axis.
+
+    >>> from scipy.stats import gstd
+    >>> a = np.arange(1, 25).reshape(2, 3, 4)
+    >>> gstd(a, axis=None)
+    2.2944076136018947
+    >>> gstd(a, axis=2)
+    array([[1.82424757, 1.22436866, 1.13183117],
+           [1.09348306, 1.07244798, 1.05914985]])
+    >>> gstd(a, axis=(1,2))
+    array([2.12939215, 1.22120169])
+    """
+    try:
+        with np.errstate(invalid='ignore'):
+            a = np.asanyarray(a)
+            # np.sum avoids creating a huge array in memory
+            all_finite = np.isfinite(a.sum())
```

Comment:
TBD if we need this. -inf is captured by ">0" and +inf probably leads to nan as an output which seems acceptable (?)